### PR TITLE
HIP-584: Copy HederaPrngSeedOperation from services

### PR DESCRIPTION
**Pull request by bilyana-gospodinova**
**Description**:
Calling block.prevrandao causes a NullPointerException. This PR adds the missing HederaPrngSeedOperation from services.

Fixes https://github.com/hashgraph/hedera-mirror-node/issues/7036

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/HederaPrngSeedOperation.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import com.hedera.services.txns.util.PrngLogic;
+import javax.inject.Inject;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.hyperledger.besu.evm.EVM;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+import org.hyperledger.besu.evm.operation.AbstractOperation;
+
+public class HederaPrngSeedOperation extends AbstractOperation {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/HederaPrngSeedOperation.java:30]: We should move this class in services package.

Under `services/evm/contracts/operations`. Then we should add a `Bean` definition in `ServicesConfiguration` for it.

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/HederaPrngSeedOperation.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    private final OperationResult successResponse;
+    private final OperationResult oogResponse;
+    private final PrngLogic prngLogic;
+
+    private final long gasCost;
+
+    @Inject
+    public HederaPrngSeedOperation(GasCalculator gasCalculator, PrngLogic prngLogic) {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/HederaPrngSeedOperation.java:39]: We can remove the `Inject` annotation when we add the Bean definition.

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/HederaPrngSeedOperation.java (hunk 1 above)
Comment (bilyana-gospodinova) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/HederaPrngSeedOperation.java]: Done

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/HederaPrngSeedOperation.java (hunk 2 above)
Comment (bilyana-gospodinova) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/HederaPrngSeedOperation.java]: Done

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/MirrorEvmTxProcessorFacadeImpl.java [hunk 3]
```diff
@@ -66,6 +67,7 @@ public class MirrorEvmTxProcessorFacadeImpl implements MirrorEvmTxProcessorFacad
     private final TraceProperties traceProperties;
     private final BasicHbarCentExchange basicHbarCentExchange;
     private final PrngSystemPrecompiledContract prngSystemPrecompiledContract;
+    private final PrngLogic prngLogic;
```
[the remark this answers is not in the document]
Comment (bilyana-gospodinova) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/MirrorEvmTxProcessorFacadeImpl.java]: Done

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/EvmOperationConstructionUtil.java [hunk 4]
```diff
@@ -76,8 +77,10 @@ public class EvmOperationConstructionUtil {
     public static final String EVM_VERSION = EVM_VERSION_0_34;
 
     public static Map<String, Provider<ContractCreationProcessor>> ccps(
-            final GasCalculator gasCalculator, final MirrorNodeEvmProperties mirrorNodeEvmProperties) {
-        final var evm = constructEvm(gasCalculator, mirrorNodeEvmProperties);
+            final GasCalculator gasCalculator,
+            final MirrorNodeEvmProperties mirrorNodeEvmProperties,
+            final HederaPrngSeedOperation prngSeedOperation) {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/execution/EvmOperationConstructionUtil.java:82]: Once we extract `ccps` and `mcps` as beans we could directly inject `HederaPrngSeedOperation` as a bean, but for now it looks OK.

File: hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/operations/HederaPrngSeedOperation.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+import com.hedera.services.txns.util.PrngLogic;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.hyperledger.besu.evm.EVM;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+import org.hyperledger.besu.evm.operation.AbstractOperation;
+
+public class HederaPrngSeedOperation extends AbstractOperation {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/operations/HederaPrngSeedOperation.java:29]: I believe this class is not moved into `services` package?

Comment (bilyana-gospodinova) [reply hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/contracts/operations/HederaPrngSeedOperation.java]: I misread the initial comment. The class is now moved to services.

**Review by steven-sheehy — APPROVED** [review]
LGTM. Can merge once Ivan re-reviews

**Pull request merged**